engine.js

```diff
@@ -122,22 +122,23 @@ async function hideThoughts() {
     const characterSettings = getCurrentCharacterSettings();
 
     const isMindReaderCharacter = Boolean(characterSettings && characterSettings.is_mind_reader);
-    const hasAccessToThought = (chatCharacter) => isMindReaderCharacter || chatCharacter.name === currentCharacter.name;
+    const hasAccessToThought = (chatCharacter) => isMindReaderCharacter || chatCharacter.thoughts_for === currentCharacter.name;
 
     let promises = [];
     const lastMessageIndex = context.chat.length - 1;
     for (let i = lastMessageIndex, thoughtsCount = []; lastMessageIndex - i < settings.max_hiding_thoughts_lookup; i--) {
+        if (i < 0) break;
```

Comment:
Because otherwise it will - and does - iterate backwards through potentially dozens of negative message indices that will never exist, which is horribly inefficient, but I can get rid of it.

Reply:
I see, you're right. Let's put this condition to the body of the for loop, like this:
```javascript
for (let i = lastMessageIndex, thoughtsCount = []; lastMessageIndex - i < settings.max_hiding_thoughts_lookup && i >= 0; i--) {
```

Reply:
Done! Also added a probably-unnecessary set of parentheses to make the logic grouping with the subtraction a bit more explicit since I don't trust JavaScript.